server/block/banner_pattern_type.go

```diff
@@ -0,0 +1,346 @@
+		BorderBannerPattern(),
+		BricksBannerPattern(),
+		CircleBannerPattern(),
+		CreeperBannerPattern(),
+		CrossBannerPattern(),
+		CurlyBorderBannerPattern(),
+		DiagonalLeftBannerPattern(),
+		DiagonalRightBannerPattern(),
+		DiagonalUpLeftBannerPattern(),
+		DiagonalUpRightBannerPattern(),
+		FlowerBannerPattern(),
+		GradientBannerPattern(),
+		GradientUpBannerPattern(),
+		HalfHorizontalBannerPattern(),
+		HalfHorizontalBottomBannerPattern(),
+		HalfVerticalBannerPattern(),
+		HalfVerticalRightBannerPattern(),
+		MojangBannerPattern(),
+		RhombusBannerPattern(),
+		SkullBannerPattern(),
+		SmallStripesBannerPattern(),
+		SquareBottomLeftBannerPattern(),
+		SquareBottomRightBannerPattern(),
+		SquareTopLeftBannerPattern(),
+		SquareTopRightBannerPattern(),
+		StraightCrossBannerPattern(),
+		StripeBottomBannerPattern(),
+		StripeCenterBannerPattern(),
+		StripeDownLeftBannerPattern(),
+		StripeDownRightBannerPattern(),
+		StripeLeftBannerPattern(),
+		StripeMiddleBannerPattern(),
+		StripeRightBannerPattern(),
+		StripeTopBannerPattern(),
+		TriangleBottomBannerPattern(),
+		TriangleTopBannerPattern(),
+		TrianglesBottomBannerPattern(),
+		TrianglesTopBannerPattern(),
+		GlobeBannerPattern(),
+		PiglinBannerPattern(),
+	}
+}
+
+type bannerPatternType uint8
+
+// Uint8 returns the bannerPatternType as a uint8.
+func (b bannerPatternType) Uint8() uint8 {
+	return uint8(b)
+}
+
+// String returns the bannerPatternType as a string.
+func (b bannerPatternType) String() string {
+	switch b {
+	case 0:
+		return "border"
+	case 1:
+		return "bricks"
+	case 2:
+		return "circle"
+	case 3:
+		return "creeper"
+	case 4:
+		return "cross"
+	case 5:
+		return "curly_border"
+	case 6:
+		return "diagonal_left"
+	case 7:
+		return "diagonal_right"
+	case 8:
+		return "diagonal_up_left"
+	case 9:
+		return "diagonal_up_right"
+	case 10:
+		return "flower"
+	case 11:
+		return "gradient"
+	case 12:
+		return "gradient_up"
+	case 13:
+		return "half_horizontal"
+	case 14:
+		return "half_horizontal_bottom"
+	case 15:
+		return "half_vertical"
+	case 16:
+		return "half_vertical_right"
+	case 17:
+		return "mojang"
+	case 18:
+		return "rhombus"
+	case 19:
+		return "skull"
+	case 20:
+		return "small_stripes"
+	case 21:
+		return "square_bottom_left"
+	case 22:
+		return "square_bottom_right"
+	case 23:
+		return "square_top_left"
+	case 24:
+		return "square_top_right"
+	case 25:
+		return "straight_cross"
+	case 26:
+		return "stripe_bottom"
+	case 27:
+		return "stripe_center"
+	case 28:
+		return "stripe_downleft"
+	case 29:
+		return "stripe_downright"
+	case 30:
+		return "stripe_left"
+	case 31:
+		return "stripe_middle"
+	case 32:
+		return "stripe_right"
+	case 33:
+		return "stripe_top"
+	case 34:
+		return "triangle_bottom"
+	case 35:
+		return "triangle_top"
+	case 36:
+		return "triangles_bottom"
+	case 37:
+		return "triangles_top"
+	case 38:
+		return "globe"
+	case 39:
+		return "piglin"
+	}
+	panic("should never happen")
```

Comment:
```suggestion
	panic("unmapped banner type")
```

Reply:
Doesn't really matter here + we should "should never happen" consistently across many areas.